config/v1/types_idp.go

```diff
@@ -6,9 +6,9 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// IdentityProvider holds cluster-wide information about IdentityProvider.  The canonical name is `cluster`
+// ClusterIdentityProvider holds cluster-wide information about IdentityProvider.  The canonical name is `cluster`
```

Comment:
@deads2k I changed this so that it didn't conflict with the `IdentityProvider` type defined in types_oauth.go. Should I just delete this file?

(I'm still conflicted about having separate resources per identity provider with some sort of ordering on them, but we decided to go with in-lining them in the main config object so I'm going with that for now)

Reply:
> @deads2k I changed this so that it didn't conflict with the `IdentityProvider` type defined in types_oauth.go. Should I just delete this file?

I thought that one was named `OAuth`.  Are you anticipating another type there?  We should keep the top level types well named for a cluster-admin, but the lower level ones are only see by users in serializations, so they can have funny names.  

I think that cluster is implied given it's location here, so I would keep it as IdentityProvider.

Reply:
I had named it `OAuthIdentityProvider` so it wouldn't conflict with this one, but that just seemed confusing: "OAuthIdentityProvider" sounds like it's for configuring OAuth Identity Providers (e.g. google, github, openid).
Also, what is this one supposed to be used for? Are you sure it isn't what I added in the oauth types?

Reply:
I'm flexible, my initial pull was simply a starting point.  There are some options specific to osin as I recall.

I think this is pretty clearly an `IdentityProvider` though.  I don't think that calling it a `ClusterIdentityProvider` is more clear.

Reply:
Who or what is using this type though? If it's not used by anyone, and no one has plans to, we can remove it.